cmd/notifications/main.go

```diff
@@ -134,3 +151,23 @@ func newCredentialsService(cfg *config.Configuration, storage *db.Storage, cache
 
 	return claimsService, nil
 }
+
+func checkDID(ctx context.Context, cfg *config.Configuration, vaultCli *api.Client) error {
+	log.Info(ctx, "Checking issuer did value", "did", cfg.APIUI.Issuer)
+	if cfg.APIUI.Issuer == "" {
+		var err error
+		cfg.APIUI.Issuer, err = providers.GetDID(ctx, vaultCli)
+		if err != nil {
+			log.Error(ctx, "cannot get issuer did from vault", "error", err)
+			return err
+		}
+		log.Info(ctx, "Issuer Did from vault", "did", cfg.APIUI.Issuer)
+		issuerDID, err := core.ParseDID(cfg.APIUI.Issuer)
+		if err != nil {
+			log.Error(ctx, "invalid issuer did format", "error", err)
+			return err
+		}
+		cfg.APIUI.IssuerDID = *issuerDID
+	}
+	return nil
+}
```

Comment:
I see this function repeated i some places. I think we could move it to a common package and reuse it. IMHO, a good place could be in config package, because it is checking an updating the configuration